sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemSerialization.java

```diff
@@ -3,53 +3,47 @@
+import com.azure.json.JsonProviders;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonToken;
 import com.azure.monitor.opentelemetry.exporter.implementation.models.TelemetryItem;
-import com.fasterxml.jackson.annotation.JsonInclude;
-import com.fasterxml.jackson.core.JsonGenerator;
-import com.fasterxml.jackson.core.io.SerializedString;
-import com.fasterxml.jackson.databind.MappingIterator;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.SerializationFeature;
 
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.io.StringWriter;
-import java.nio.ByteBuffer;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.zip.GZIPInputStream;
-import java.util.zip.GZIPOutputStream;
 
 public final class TelemetryItemSerialization {
 
-    private static final ObjectMapper mapper = createObjectMapper();
-    private static final AppInsightsByteBufferPool byteBufferPool = new AppInsightsByteBufferPool();
-    private static final ClientLogger logger = new ClientLogger(TelemetryItemSerialization.class);
-
-    public static List<TelemetryItem> deserialize(byte[] data) {
-        return deserializeAlreadyDecoded(decode(data));
-    }
-
     // visible for testing
-    // deserialize raw bytes to a list of TelemetryItem without decoding
-    public static List<TelemetryItem> deserializeAlreadyDecoded(byte[] data) {
-        try {
-            MappingIterator<TelemetryItem> iterator = mapper.readerFor(TelemetryItem.class).readValues(data);
-            return iterator.readAll();
-        } catch (Exception e) {
-            throw new IllegalStateException("Failed to deserialize byte[] to a list of TelemetryItems", e);
+    // deserialize multiple TelemetryItem raw bytes with newline delimiters to a list of TelemetryItems
+    public static List<TelemetryItem> deserialize(byte[] rawBytes) {
+        try (JsonReader jsonReader = JsonProviders.createReader(rawBytes)) {
+            JsonToken token = jsonReader.currentToken();
+            if (token == null) {
+                token = jsonReader.nextToken();
+            }
+
+            List<TelemetryItem> result = new ArrayList<>();
+            do {
+                result.add(TelemetryItem.fromJson(jsonReader));
+            } while (jsonReader.nextToken() == JsonToken.START_OBJECT);
+            return result;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
         }
     }
 
-    // decode gzipped request raw bytes back to original request raw bytes
-    private static byte[] decode(byte[] rawBytes) {
-        try (GZIPInputStream in = new GZIPInputStream(new ByteArrayInputStream(rawBytes))) {
-            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    // visible for testing
+    // un-gzip TelemetryItems raw bytes back to original un-gzipped TelemetryItems raw bytes
+    public static byte[] ungzip(byte[] rawBytes) {
```

Comment:
looks like can move to `LocalStorageTelemetryPipelineListener` and make package-protected over there (then we can delete this class!)